File: android2_lesson04/src/java/java/com/example/ama/android2_lesson04/ui/viewer/PVFragmentStartedService.java
package com.example.ama.android2_lesson04.ui.viewer;

import android.content.Intent;

import com.example.ama.android2_lesson04.ServiceTestApp;
import com.example.ama.android2_lesson04.background.service.LoadPictureStartedService;
import com.example.ama.android2_lesson04.background.service.ServiceConstants;
import com.example.ama.android2_lesson04.ui.viewer.base.BaseReceiverFragment;
import com.example.ama.android2_lesson04.utils.ResourcesUtils;

public class PVFragmentStartedService extends BaseReceiverFragment {

    public static PVFragmentStartedService newInstance() {
        return new PVFragmentStartedService();
    }

    @Override
    protected void onLoadImagesClick() {
        startMyService();
    }

    private void startMyService() {
        if (getActivity() != null) {
            Intent intent = new Intent(ServiceTestApp.getInstance(), LoadPictureStartedService.class);
            intent.putExtra(ServiceConstants.EXTRA_KEY, ResourcesUtils.getUrlsArray());
            getActivity().startService(intent);
        }
    }
}

File: android2_lesson04/src/kotlin/java/com/example/ama/android2_lesson04/ui/viewer/PVFragmentStartedService.kt
package com.example.ama.android2_lesson04.ui.viewer

import android.content.Intent
import com.example.ama.android2_lesson04.R
import com.example.ama.android2_lesson04.ServiceTestApp
import com.example.ama.android2_lesson04.background.service.LoadPictureStartedService
import com.example.ama.android2_lesson04.ui.viewer.base.BaseReceiverFragment
import com.example.ama.android2_lesson04.ui.viewer.base.EXTRA_KEY

class PVFragmentStartedService : BaseReceiverFragment() {

    companion object {
        fun newInstance() = PVFragmentStartedService()
    }

    override fun onLoadImagesClick() {
        startService()
    }

    private fun startService() {
        val intent = Intent(ServiceTestApp.instance, LoadPictureStartedService::class.java)
        intent.putExtra(EXTRA_KEY, resources.getStringArray(R.array.image_urls))
        activity?.startService(intent)
    }
}